Dockerfile

```diff
@@ -28,8 +28,22 @@ COPY --from=builder /src/webapp /app
 # through `envsubst` by the nginx docker image entry point.
 COPY /docker/nginx-templates/* /etc/nginx/templates/
 
+# Override main nginx config, to make it suitable for use with non-root user
+RUN sed -i \
+      -e '/user *nginx;/d' \
+      -e  's,/var/run/nginx.pid,/tmp/nginx.pid,' \
+      -e "/^http {/a \    proxy_temp_path /tmp/proxy_temp;\n    client_body_temp_path /tmp/client_temp;\n    fastcgi_temp_path /tmp/fastcgi_temp;\n    uwsgi_temp_path /tmp/uwsgi_temp;\n    scgi_temp_path /tmp/scgi_temp;\n" \
+      /etc/nginx/nginx.conf
```

Comment:
Just a thought: would it be better at this point to just replace the nginx.conf with our own? Then we don't have to wrap our brains around the sed, and less risk of it breaking if the config file changes in later images versions. It's not a huge file. Also, do we still need the fastcgi / uwsgi / scgi stuff or can we turn it off entirely?

Reply:
tbh my thinking was the opposite: what if something important turns up in the base nginx.conf in a future version.

I don't disagree though. I'll have a go.

Reply:
Since:
 * the `user` directive is just ignored when running as non-root
 * we don't use any of the proxy/fastcgi/uwsgi/scgi stuff
 * we  don't have a usecase for client body uploads

... the only thing we actually need to change here is the pidfile. 

Given that, I'm minded to retain the sed incantation, greatly simplified.

Reply:
Fair, that's an amount of sed that my brain can deal with!